packages/fxa-payments-server/server/lib/server.js

```diff
@@ -153,6 +154,17 @@ module.exports = () => {
   );
 
   if (config.get('csp.enabled')) {
+    app.use(function (req, res, next) {
+      // Generate nonce for CSP to allow paypal inline script.
+      res.paypalCspNonce = uuid();
+      next();
+    });
+
+    // Add nonce for paypal's inline script.
+    cspRulesBlocking.directives.scriptSrc.push((req, res) => {
+      return `'nonce-${res.paypalCspNonce}'`;
```

Comment:
Not knowing express intimately and just from looking at this, I don't actually know if the function passed into `app.use` on line 157 has executed by the time when this line executes.  It does appear to work.  Is there doc where I can find this info?

Reply:
`app.use()` basically adds middleware to the request handling. They will execute in the order they are declared. So, the middleware that sets the nonce above will run right before the csp middleware which is set up below on line 168.

From https://expressjs.com/en/guide/writing-middleware.html:

> The order of middleware loading is important: middleware functions that are loaded first are also executed first.

Reply:
Right, that I do remember.  But `cspRulesBlocking.directives.scriptSrc.push` is not `app.use`.  Now that I re-read that bit about middleware order of execution, I see why it is working.  👍